library/src/domain/Co2Eq/Co2Eq.ts

```diff
@@ -0,0 +1,111 @@
+import * as Either from "fp-ts/lib/Either";
+import { pipe } from "fp-ts/lib/function";
+import * as Codec from "io-ts/lib/Codec";
+import * as Decoder from "io-ts/lib/Decoder";
+import { Co2EqUnit, Co2EqUnitEnum } from "./Co2EqUnit";
+
+export class Co2Eq {
+  averageInGrams;
+
+  constructor({ averageValue, unit }: Co2EqProps) {
+    this.averageInGrams = convertToGrams(averageValue, unit);
+  }
+
+  format(co2EqUnit?: Co2EqUnitEnum) {
+    const unit = co2EqUnit
+      ? co2EqUnit
+      : findBestUnitForAmountInGrams(this.averageInGrams);
+
+    switch (unit) {
+      case Co2EqUnitEnum.KILOGRAM:
+        return `${this.averageInGrams / 1e3} kg`;
+
+      case Co2EqUnitEnum.TON:
+        return `${this.averageInGrams / 1e6} tons`;
```

Comment:
```suggestion
        return `${this.averageInGrams / 1e6} ton`;
```

I think, because the parser don't except a `s`

Reply:
true, and I think the parser should accept an s.